src/web/utils/cursorUtils.ts

```diff
@@ -54,28 +54,21 @@ function setCursorPosition(target: MarkdownTextInputElement, startIndex: number,
 
 function scrollIntoView(target: MarkdownTextInputElement, node: TreeNode) {
   const targetElement = target;
-  if (node.type === 'br' && node.parentNode?.parentNode?.type === 'line') {
-    // If the node is a line break, scroll to the parent paragraph, because Safari doesn't support scrollIntoView on br elements
-    node.parentNode.parentNode.element.scrollIntoView({
-      block: 'nearest',
-    });
-  } else {
-    const selection = window.getSelection();
-    if (selection && selection.rangeCount > 0) {
-      const range = selection.getRangeAt(0);
-      const caretRect = range.getBoundingClientRect();
-      const targetRect = target.getBoundingClientRect();
-
-      // In case the caret is below the visible input area, scroll to the end of the node
-      if (caretRect.top + caretRect.height > targetRect.top + targetRect.height) {
-        targetElement.scrollTop = caretRect.top + caretRect.height - targetRect.top - targetRect.height + target.scrollTop;
-      }
-    }
-
-    node.element.scrollIntoView({
-      block: 'nearest',
-    });
+  const orderIndex = Number(node.orderIndex.split(',')[0]);
+  const currentLine = target.tree.childNodes[orderIndex]?.element;
+  const scrollTargetElement = currentLine || node.element;
+
+  const caretRect = scrollTargetElement.getBoundingClientRect();
+  const targetRect = target.getBoundingClientRect();
+  // In case the caret is below the visible input area, scroll to the end of the node
+  if (caretRect.top + caretRect.height > targetRect.top + targetRect.height) {
+    targetElement.scrollTop = caretRect.top - targetRect.top + target.scrollTop - targetRect.height + caretRect.height + 4;
+    return;
```

Comment:
Do we need this return?

Reply:
Yes, we only want to change scrollTop. Executing the rest of the code below can cause scroll jumps